logistics.lua

```diff
@@ -19,26 +21,32 @@ function logistics.save_logistic_layout(player, name)
     util.log('Saving current layout as ' .. name)
     global.layouts[name] = {
         slots = slots,
-        slot_count = player.character_logistic_slot_count,
+        slot_count = player.character.request_slot_count,
     }
 end
 
 function logistics.clear_logistic_layout(player)
-    for i = 1, player.character_logistic_slot_count do
+	if not check_player_has_character(player) then return end
+	
+
+    for i = 1, player.character.request_slot_count do
         player.clear_personal_logistic_slot(i)
     end
-    player.character_logistic_slot_count = 0
+    -- player.character.request_slot_count = 0
+	-- it's probably not necessary anymore
```

Comment:
if this is because the slot count now increments automatically and this isn't needed, let's remove the code instead of commenting it out